core/src/main/java/io/questdb/cairo/TxReader.java

```diff
@@ -51,23 +51,22 @@
     protected int partitionBy;
     protected long partitionTableVersion;
     protected int attachedPartitionsSize = 0;
+    private PartitionBy.PartitionFloorMethod partitionFloorMethod;
     private MemoryMR roTxMem;
 
-    public TxReader(FilesFacade ff, @Transient Path path, int partitionBy) {
-        try {
-            roTxMem = openTxnFile(ff, path);
-            this.partitionFloorMethod = PartitionBy.getPartitionFloorMethod(partitionBy);
-            this.partitionBy = partitionBy;
-        } catch (Throwable e) {
-            close();
-            throw e;
-        }
+    public TxReader(FilesFacade ff) {
+        this.ff = ff;
     }
 
     public boolean attachedPartitionsContains(long ts) {
         return findAttachedPartitionIndex(ts) > -1;
     }
 
+    @Override
+    public void clear() {
+        close();
```

Comment:
nit: do we need this method considering that it's the same as `close()`?

Reply:
Just to implement `Mutable` interface